internal/hsm/hsm_test.go

```diff
@@ -0,0 +1,328 @@
+// MIT License
```

Comment:
This is going to cause a conflict, since my PR has this file already.

Reply:
I made note in my PR that yours should get merged 1st. The code in internal/hsm is an exact copy of the code from your PR as of this morning plus some additions I needed for power capping. The conflicts shouldn't be too bad.

Reply:
I mark this comment resolved once your PR has merged to develope and I've dealt with the conflicts

Reply:
I've updated this PR with develop after Matt's HSM layer was merged